File: Prototype/BigGp/src/Model/BaseNodes/Assignment/FunctionCallIn.java
package Model.BaseNodes.Assignment;

import Model.Node;
import Model.Interfaces.Terminal;
import Serialization.Token;

import java.util.ArrayList;

public class FunctionCallIn extends Node implements Terminal {

    private final String value;

    public ArrayList<Node> getChildrenAsNodes(){
        ArrayList<Node> nodes = new ArrayList<>();
        nodes.add(this);

        return nodes;
    }

    @Override
    public String toString() {
        return value;
    }

    @Override
    public String getValue() {
        return value;
    }

    public FunctionCallIn(Node parentNode) {
        super(parentNode, "FunctionCallIn", true);
        this.value = "read()";
    }

    public FunctionCallIn(Node parent, ArrayList<Token> tokens) {
        super(parent, "FunctionCallIn", true);

        this.value = tokens.remove(0).getVALUE();
    }
}

File: Interactive.Genetic.Programming.Environment/src/Domain/Model/Nodes.Big/Assignments/FunctionCallIn.cs
using Model.Abstract;
using Model.Extensions;
using Model.Interfaces;

namespace Model.Nodes.Big.Assignments;

public sealed class FunctionCallIn : Node, ITerminal
{
    public string Value { get; }

    public override List<Node> ChildrenAsNodes() => [this];

    public FunctionCallIn(Node parentNode) : base(parentNode, "FunctionCallIn", true)
    {
        Value = "read()";
    }
    
    public FunctionCallIn(Node parentNode, IList<Token> tokens) : base(parentNode, "FunctionCallIn", true)
    {
        Value = tokens.PopFront().Value ?? throw new Exception("Token value cannot be null");
    }
    
    public override string ToString() => Value;
}